extensions/oplogPopulator/modules/Connector.js

```diff
@@ -109,12 +110,25 @@ class Connector {
         }
     }
 
+    /**
+     * Updates partition name in connector config
+     * @returns {undefined}
+     */
+    updatePartitionName() {
+        this._config['offset.partition.name'] = `partition-${uuid.v4()}`;
+    }
+
     /**
      * Creates the Kafka-connect mongo connector
      * @returns {Promise|undefined} undefined
      * @throws {InternalError}
      */
     async spawn() {
+        if (this._isRunning) {
```

Comment:
this actually never happens, it is just an extra safety check?
→ should log an error if it happens

Reply:
yes just a safety check